### DISCUSSION TITLE
Bundle UpgradeCode guid variable attribute is invalid?

### DISCUSSION THREAD

**Discussion by wstaelens** [wixtoolset/issues#7597] [Questions] [answered]
I have two projects (1) a Wix bootstrapper and (2) an Wix Installer for msi.

In the Installer I have an include file (Foo.wxi) which has:

``` 
<?xml version="1.0" encoding="utf-8"?>
<Include xmlns="http://wixtoolset.org/schemas/v4/wxs">
	<?define UpgradeCode="{117df273-f116-4c31-8f2d-c84fae5aef2a}" ?>
</Include>
```

In my bootstrapper wix project I include in my Bundle.wxs the file Foo.wxi which exists in my other Wix Installer project:

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" 
	 xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal"
	 xmlns:netfx="http://wixtoolset.org/schemas/v4/wxs/netfx">

	<?include $(Installer.ProjectDir)Foo.wxi?>
```

And I try to use the variable in the Bundle-element:

```
<Bundle UpgradeCode="$(UpgradeCode)">
<!-- ... -->
``` 
Visual Studio 2022 tells me that the attribute is invalid:
![image](https://github.com/wixtoolset/issues/assets/7447285/c2ba51fb-b5df-4119-9142-d9cc1b5bba4e)

When I change the syntax from `$(UpgradeCode)` to `$(var.UpgradeCode)` Visual Studio does not complain:

```
<Bundle UpgradeCode="$(var.UpgradeCode)">
<!-- ... -->
``` 

![image](https://github.com/wixtoolset/issues/assets/7447285/5019c0aa-25e2-4641-8747-0b06df7a6237)

I tried also these with or without curly brackets (not sure what the difference it makes but it didn't help):
```
<?define UpgradeCode="{117df273-f116-4c31-8f2d-c84fae5aef2a}" ?>
<?define UpgradeCode="117df273-f116-4c31-8f2d-c84fae5aef2a" ?>
```

I thought that the syntax in Wix v4 was `$(abc)` now instead of `$(var.abc)`.  Is this a bug or am I doing something wrong?

Both of my projects use 4.0.1: `<Project Sdk="WixToolset.Sdk/4.0.1">`.

(Created a discussion as I am not a Wix expert to know if this is an issue or if I'm doing something incorrect).

**Comment by barnson** [comment, marked as answer]
It's a bug in the .xsd -- `var.`-less should be validated. That said, it only affects Intellisense inside VS.

**Reply by wstaelens** [reply]
ok, i'll leave it out to you to create an issue in case this is an .xsd from wix/wixtoolset. As I have no clue on how/where these are.

**Comment by wstaelens** [comment]
Just saying that this is still the case in v4 (4.0.5).

**Reply by robmen** [reply]
Maybe your apathy didn't inspire action? :smile:

**Reply by wstaelens** [reply]
(I came across my own ticket while searching why Visual Studio intellisense reported this as incorrect. I just thought to inform that this is still true in version 4.0.5, nothing more.)

**Reply by robmen** [reply]
Ahh, sorry. My response was based off your last comment on the thread.

**Reply by wstaelens** [reply]
I understand (P.S. I still didn't find that .xsd file 🙃  )

**Reply by robmen** [reply]
XSDs are here. https://github.com/wixtoolset/web/tree/master/src/xsd4
